File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.cc
#include <iostream>

using namespace std;

int main() {
    int n,m;
    cin >> n >> m;
    bool ice[205][205];
    for(int i=0;i<m;i++)
    {
        int x,y;
        cin >> x >> y;
        ice[x][y] = ice[y][x] = true;
    }
    int ans = 0;
    for(int i=1;i<=n;i++)
    {
        for(int j=i+1;j<=n;j++)
        {
            if(ice[i][j]) continue;
            for(int k=j+1;k<=n;k++)
            {
                if(ice[i][k]||ice[j][k]) continue;
                ans++;
            }
        }
    }
    cout<<ans;
}

File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt(),m = s.nextInt();
      boolean[][] ice = new boolean[205][205];
      
      for(int i=0;i<m;i++)
      {
          int x = s.nextInt(),y = s.nextInt();
          ice[x][y] = true;
          ice[y][x] = true;
      }
      int ans = 0;
      for(int i=1;i<=n;i++)
      {
          for(int j=i+1;j<=n;j++)
          {
              if(ice[i][j]) continue;
              for(int k=j+1;k<=n;k++)
              {
                  if(ice[i][k]||ice[j][k])continue;
                  ans++;
              }
          }
      }
      System.out.println(ans);
    }
}